pyproject.toml

```diff
@@ -87,53 +76,204 @@ dev = [
     "livereload",
     "pillow",
     "cairosvg",
-    ## Constrain system
+
+]
+# Development tools (testing via hatch test)
+dev = [
+    "compliance-trestle[typing,docs]",
+    "pre-commit>=2.4.0",
+    "python-semantic-release>=10.4.0",
     "pylint",
-    # Checking repo after docs generation.
-    "gitpython"
+    "datamodel-code-generator[http] >= 0.36.0",
+    "gitpython",
 ]
 
 [project.urls]
 Homepage = "https://oscal-compass.github.io/compliance-trestle"
+Documentation = "https://oscal-compass.github.io/compliance-trestle"
+"Source code" = "https://github.com/oscal-compass/compliance-trestle"
 
 [project.scripts]
 trestle = "trestle.cli:run"
 
-[tool.setuptools]
-include-package-data = true
+[tool.hatch.version]
+path = "trestle/__init__.py"
+
+[tool.hatch.build.targets.sdist]
+include = [
+    "/trestle",
+]
+exclude = [
+    "/tests",
+]
+
+[tool.hatch.build.targets.wheel]
+packages = ["trestle"]
+
+[tool.hatch.envs.hatch-test]
+default-args = ["tests"]
+parallel = true
+randomize = true
+# Pin coverage for Python 3.9 compatibility (7.7+ requires Python 3.10+)
+dependencies = [
+    "coverage[toml]~=7.4",
+    "pytest>=8.1",
+    "pytest-xdist[psutil]>=3.5",
+    "pytest-randomly>=3.15", 
+    "mypy"
+]
 
-[tool.setuptools.packages.find]
-include = ["trestle*"]
-exclude = ["tests"]
+[[tool.hatch.envs.hatch-test.matrix]]
+python = ["3.9", "3.10", "3.11", "3.12"]
 
```

Comment:
@butler54 The min and max python versions are now in a separate file from where it is read. We don't run tests on all versions of python (only min and max).